pkg/cmd/server/start/master_args.go

```diff
@@ -149,7 +154,12 @@ func (args MasterArgs) BuildSerializeableMasterConfig() (*configapi.MasterConfig
 	// always include localhost as an allowed CORS origin
 	// always include master public address as an allowed CORS origin
 	corsAllowedOrigins := sets.NewString(args.CORSAllowedOrigins...)
-	corsAllowedOrigins.Insert(assetPublicAddr.Host, masterPublicAddr.Host, "localhost", "127.0.0.1")
+	corsAllowedOrigins.Insert(
+		makeExactMatchRegex(assetPublicAddr.Host),
+		makeExactMatchRegex(masterPublicAddr.Host),
+		makeExactMatchRegex("localhost"),
+		makeExactMatchRegex("127.0.0.1"),
```

Comment:
It looks like this will now block origins like `https://localhost:9000`? If so, I wonder if it's still worth settings these as defaults.

Reply:
If we trust localhost, I assume we trust it on any port and protocol?

Reply:
Do `assetPublicAddres.Host` and `masterPublicAddres.Host` include scheme and port as well?

Reply:
hmm, yeah... need to rethink the scheme/port bit

Reply:
ok, pinned to the beginning of the host with `//` and the end of the host with `(:|$)` or `$`, depending on whether host already contained a port.

This results in output like this:
```
corsAllowedOrigins:
- //127\.0\.0\.1(:|$)
- //192\.168\.1\.101:8443$
- //localhost(:|$)
```

@spadgett, are you able to check if that works across browsers for the cross-domain case?